src/Kestrel.Core/Internal/Http/HttpProtocol.cs

```diff
@@ -1343,7 +1343,7 @@ private Pipe CreateRequestBodyPipe()
             (
                 pool: _context.MemoryPool,
                 readerScheduler: ServiceContext.Scheduler,
-                writerScheduler: PipeScheduler.Inline,
+                writerScheduler: PipeScheduler.ThreadPool,
```

Comment:
This being inline is fine for HTTP/1.1 where there's no multiplexing. We should make this virtual, and only change the HTTP/2 behavior.

Reply:
Also this affects the scheduling of server code, not app code, so I don't see how this could affect whether or not the app code could block the connection loop.